src/dnslib/dns.py

```diff
@@ -0,0 +1,3341 @@
+"""
+    DNS - main dnslib module
+
+    Contains core DNS packet handling code
+"""
+
+
+import base64
+import binascii
+import calendar
+import collections
+import copy
+from ipaddress import IPv4Address, IPv6Address
+from itertools import chain, zip_longest
+import os.path
+import random
+import socket
+import string
+import struct
+import sys
+import textwrap
+import time
+from typing import overload, Type, TypeVar, Optional, Union, List, Tuple, Dict, Any, Sequence, Generator, cast
+
+# Sequence deprecated in 3.9, keep track in case it is removed.
```

Comment:
Sequence is probably not what we want as a `str` is still a `Sequence[str]`. Might be better to have a type alias for `Union[List[str],Tuple[str,...]]`